### DISCUSSION TITLE
Quasar v2 Beta, How to implement websocket?

### DISCUSSION THREAD

**Discussion by DrakoPOD** [quasarframework/quasar#8804] [General - Components / Directives / etc] [answered]
Hellow, I'm triying to use websocket, this library. [vue-native-websocket-vue3](https://github.com/likaia/vue-native-websocket-vue3).

But I new with quasar, just one week, I can't follow the example in readme, the structure of the file is different.

**Comment by metalsadman** [comment, marked as answer]
Haven't used that library but typically you initialize plugins in a boot file https://next.quasar.dev/quasar-cli/boot-files#introduction.

**Reply by DrakoPOD** [reply]
Thanks, But now I have problems with state and mutations:

```
export default async ({ app, store }) => {
  app.use(VueNativeSock,"ws://localhost:8000",{
    store: store,
  });
}
```

`[vuex] unknown mutation type: SOCKET_ONOPEN`

Is an error in this line:
`this.store[method](target, msg);`
method = "commit"

is this the correct way to call a mutation?

**Reply by DrakoPOD** [reply]
I already solved this issue
